test/test_spath.c

```diff
@@ -0,0 +1,361 @@
+  sp = spath_from_str("Directory/File/1/2/3/4");
+  rc = spath_slice(sp, 2,4);
+//  spath_slice(sp, -6,-17);
+  spath_strcpy(buff,100, sp);
+  printf("slice result is %s\n",buff);
+  if(rc != SPATH_SUCCESS || strcmp(buff, "1/2/3/4") != 0 ){
+    printf ("Error in line %d, file %s, function %s.\n", __LINE__, __FILE__, __func__);
+    printf("test_spath failed, spath_slice() failed \n");
+    return TEST_FAIL;
+  }
+
+//This test has been invalidated by Adam Moody's changes to functionality in Pull Request 6 on Github.
+//  spath_delete(&sp);
+//  sp = spath_from_str("Directory/File/1/2/3/4");
+//  rc = spath_slice(sp,2,5);
+//  if(rc == SPATH_SUCCESS){
+//    printf ("Error in line %d, file %s, function %s.\n", __LINE__, __FILE__, __func__);
+//    printf("test_spath failed, spath_slice() returned SPATH_SUCCESS when (offset + length) > components \n");
+//    return TEST_FAIL;
+//  }
+
+//This test has been invalidated by Adam Moody's changes to functionality in Pull Request 6 on Github.
+//  spath_delete(&sp);
+//  sp = spath_from_str("Directory/File/1/2/3/4");
+//  rc = spath_slice(sp,6,1);
+//  if(rc == SPATH_SUCCESS){
+//    printf ("Error in line %d, file %s, function %s.\n", __LINE__, __FILE__, __func__);
+//    printf("test_spath failed, spath_slice() returned SPATH_SUCCESS when starting index not in range \n");
+//    return TEST_FAIL;
+//  }
+
+  spath_delete(&sp);
+  sp = spath_from_str("Dir1/Dir2/Dir3/File");
+  spath_dirname(sp);
+  spath_strcpy(buff,100, sp);
+  if(strcmp(buff, "Dir1/Dir2/Dir3") != 0 ){
+    printf ("Error in line %d, file %s, function %s.\n", __LINE__, __FILE__, __func__);
+    printf("test_spath failed, spath_dirname() failed \n");
+    return TEST_FAIL;
+  }
+
+  spath_delete(&sp);
+  sp = spath_from_str("Dir1/Dir2/Dir3/File");
+  spath_basename(sp);
+  spath_strcpy(buff,100, sp);
+  if(strcmp(buff, "File") != 0 ){
+    printf ("Error in line %d, file %s, function %s.\n", __LINE__, __FILE__, __func__);
+    printf("test_spath failed, spath_basename() failed \n");
+    return TEST_FAIL;
+  }
+
+  spath_delete(&sp);
+  spath_delete(&sp1);
+  sp = spath_from_str("Directory/File/1/2/3/4");
+  sp1 = spath_sub(sp,2,4);
+//  sp1 = spath_sub(sp, -6,-17);
+  spath_strcpy(buff,100, sp1);
+  printf("sub result is %s\n",buff);
+  if(strcmp(buff, "1/2/3/4") != 0 ){
+    printf ("Error in line %d, file %s, function %s.\n", __LINE__, __FILE__, __func__);
+    printf("test_spath failed, spath_sub() returned wrong path \n");
+    return TEST_FAIL;
+  }
+
+//This test has been invalidated by Adam Moody's changes to functionality in Pull Request 6 on Github.
+//  spath_delete(&sp);
+//  spath_delete(&sp1);
+//  sp = spath_from_str("Directory/File/1/2/3/4");
+//  sp1 = spath_sub(sp,2,5);
+//  if(sp1 != NULL){
+//    printf ("Error in line %d, file %s, function %s.\n", __LINE__, __FILE__, __func__);
+//    printf("test_spath failed, spath_sub() returned NOT NULL when (offset + length) > components \n");
+//    return TEST_FAIL;
+//  }
```

Comment:
Let's update this case to also expect "1/2/3/4"

Reply:
done